beta/nncf/tensorflow/sparsity/rb/algorithm.py

```diff
@@ -0,0 +1,159 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import tensorflow as tf
+from tensorflow.python.keras.utils.layer_utils import count_params
+
+from nncf.common.graph.transformations.commands import TransformationPriority
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.common.sparsity.schedulers import SPARSITY_SCHEDULERS
+from beta.nncf.tensorflow.algorithm_selector import TF_COMPRESSION_ALGORITHMS
+from beta.nncf.tensorflow.api.compression import TFCompressionAlgorithmBuilder
+from beta.nncf.tensorflow.graph.transformations.commands import TFInsertionCommand
+from beta.nncf.tensorflow.graph.transformations.commands import TFLayerWeight
+from beta.nncf.tensorflow.graph.utils import collect_wrapped_layers
+from beta.nncf.tensorflow.graph.utils import get_original_name_and_instance_index
+from beta.nncf.tensorflow.graph.converter import convert_keras_model_to_nxmodel
+from beta.nncf.tensorflow.sparsity.base_algorithm import BaseSparsityController
+from beta.nncf.tensorflow.sparsity.base_algorithm import SPARSITY_LAYERS
+from beta.nncf.tensorflow.sparsity.rb.loss import SparseLoss
+from beta.nncf.tensorflow.sparsity.rb.operation import RBSparsifyingWeight, OP_NAME
+from beta.nncf.tensorflow.sparsity.rb.functions import binary_mask
+from beta.nncf.tensorflow.sparsity.utils import apply_fn_to_op_weights
+from beta.nncf.tensorflow.utils.node import is_ignored
+
+
+@TF_COMPRESSION_ALGORITHMS.register('rb_sparsity')
+class RBSparsityBuilder(TFCompressionAlgorithmBuilder):
+    def __init__(self, config):
+        if isinstance(tf.distribute.get_strategy(), tf.distribute.MirroredStrategy):
+            raise Exception('RB sparsity algorithm do not support the distributed mode with mirrored strategy')
+        super().__init__(config)
+        self.ignored_scopes = self.config.get('ignored_scopes', [])
+
+    def get_transformation_layout(self, model):
+        nxmodel = convert_keras_model_to_nxmodel(model)
+        transformations = TransformationLayout()
+        shared_nodes = set()
+
+        for node_name, node in nxmodel.nodes.items():
+            original_node_name, _ = get_original_name_and_instance_index(node_name)
+            if node['type'] not in SPARSITY_LAYERS\
+                    or is_ignored(node_name, self.ignored_scopes) \
+                    or original_node_name in shared_nodes:
+                continue
```

Comment:
Can you please refactor this according to the style guide?

Reply:
Done